src/ucs/stats/stats.c

```diff
@@ -625,30 +641,31 @@ static size_t ucs_stats_aggregate_sum_recurs(ucs_stats_node_t *node,
     return n_counters;
 }
 
-void ucs_stats_aggregate_sum(
+size_t ucs_stats_aggregate(
         ucs_stats_counter_t *aggregate_sum_counters,
-        size_t aggregate_sum_counters_size, size_t *n_counters_out,
+        size_t aggregate_sum_counters_size,
         const ucs_stats_aggrgt_sum_clsid_to_name_db_t **clsid_to_name_db)
 {
-    size_t n_counters            = 0;
-    ucs_stats_children_sel_t sel = UCS_STATS_ACTIVE_CHILDREN;
+    size_t n_counters = 0;
+    uint32_t i;
 
     /* Zero output counters */
-    size_t total_size_bytes = sizeof(*aggregate_sum_counters) *
-                              aggregate_sum_counters_size;
-    memset(aggregate_sum_counters, 0x00, total_size_bytes);
+    for (i = 0; i < aggregate_sum_counters_size; i++) {
+        aggregate_sum_counters[i] = 0;
+    }
 
     /* Aggregate-sum counters */
     n_counters = ucs_stats_aggregate_sum_recurs(&ucs_stats_context.root_node,
-                                                sel, n_counters,
+                                                UCS_STATS_ACTIVE_CHILDREN,
```

Comment:
n_counters is not really used by this function. its value is ignored and you always return ucs_stats_context.aggrgt_sum_db_last_offset

Reply:
Removed